src/nccl_ofi_rdma.c

```diff
@@ -303,6 +303,22 @@ static inline nccl_net_ofi_ep_rail_t *get_rail(nccl_net_ofi_rdma_ep_t *ep,
 	return &ep->rails[rail_id];
 }
 
+/*
+ * @brief return the domain for the endpoint and rail.
+ */
+
+static int get_domain(nccl_net_ofi_ep_t *base_ep, int rail, struct fid_domain **domain)
```

Comment:
I saw Amedeo touched on this, but why does `get_domain` return an int? Do you ever expect it to fail?

Reply:
+1; this feels like lots of unnecessary error checking.